File: src/main/java/net/dontdrinkandroot/gitki/service/user/UserService.java
package net.dontdrinkandroot.gitki.service.user;

import net.dontdrinkandroot.gitki.model.User;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.Iterator;
import java.util.Optional;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public interface UserService extends UserDetailsService
{
    long findCount();

    Iterator<User> find(long first, long count, String property, boolean ascending);

    boolean hasAdminUser();

    User save(User user, String password);

    void removeAll();

    void remove(User user);

    Optional<User> find(Long id);
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/service/user/UserService.kt
package net.dontdrinkandroot.gitki.service.user

import net.dontdrinkandroot.gitki.model.User
import org.springframework.security.core.userdetails.UserDetailsService
import java.util.*

interface UserService : UserDetailsService {

    fun findCount(): Long
    fun find(first: Long, count: Long, property: String, ascending: Boolean): Iterator<User>
    fun hasAdminUser(): Boolean
    fun save(user: User, password: String?): User
    fun removeAll()
    fun remove(user: User)
    fun find(id: Long): Optional<User>
}